### ISSUE TITLE
iOS a11y bridge should respect `hasImplicitScrolling`

### ISSUE THREAD

**Issue by dnfield** [flutter/flutter#44660]
Android does this - right now, per @goderbauer, iOS exclusively looks at cacheExtent instead of also respecting hasImplicitScrolling.

It should be possible to set a cache extent on something but also set `hasImplicitScrolling` to false and have the iOS bridge respect that.

**Comment by dnfield** [comment]
Talked with @goderbauer about this one.  It doesn't seem possible for iOS's bridge to use implicit scrolling flags the same way as android without there being a cache extent to actually have the node available to the bridge.  Perhaps if anything we should consider making Android _not_ work unless the cache extent is specified, for consistency?

**Comment by goderbauer** [comment]
I don't think we should make the experience on Android worse (we don't want to implement the lowest common denominator). I think iOS is the outlier here since they don't have an API for implicit scrolling.

**Comment by GaryQian** [comment]
Is there anything to do here? It looks like there isn't a way to support this on iOS?

**Comment by dnfield** [comment]
We could work around it the way that Safari does on the web, with a "rude" announcement instead of a polite one.  We'd have to directly call the a11y service to do it.

**Comment by goderbauer** [comment]
@dnfield Not sure I understand your comment about rude announcements. How would they help?

We currently enable implicit scrolling by telling iOS about all SemanticsNodes that are marked hidden at the edges of the viewport. One could imagine that we don't tell iOS about these nodes if the surrounding scrollable has `hasImplicitScrolling` set to false. That would shut off implicit scrolling on iOS as well.

**Comment by dnfield** [comment]
Sorry, I mixed up issues.  I was thinking about live regions on iOS.  

I think what's left to do here is that if implicit scrolling is set to false, we honor that on iOS.
